duffel_api/models/order_cancellation.py

```diff
@@ -22,12 +34,32 @@ class OrderCancellation:
     class InvalidRefundType(Exception):
         """Invalid refund type provided"""
 
-    def __init__(self, json):
-        for key in json:
-            value = maybe_parse_date_entries(key, json[key])
-            if (
-                key == "refund_to"
-                and value not in OrderCancellation.allowed_refund_types
-            ):
-                raise OrderCancellation.InvalidRefundType(value)
-            setattr(self, key, value)
+    def __post_init__(self):
+        if self.refund_to not in OrderCancellation.allowed_refund_types:
+            raise OrderCancellation.InvalidRefundType(self.refund_to)
+
+        return self
+
+    @classmethod
+    def from_json(cls, json: dict):
+        """Construct a class instance from a JSON response."""
+        return cls(
+            id=json["id"],
+            order_id=json["order_id"],
+            live_mode=json["live_mode"],
+            expires_at=parse_datetime(json["confirmed_at"]),
+            refund_amount=json["refund_amount"],
+            refund_currency=json["refund_currency"],
+            refund_to=json["refund_to"],
+            confirmed_at=parse_datetime(json["confirmed_at"]),
+            created_at=datetime.strptime(json["created_at"], "%Y-%m-%dT%H:%M:%S.%fZ"),
```

Comment:
Is this meant to use the new `parse_datetime` function (below)?

Reply:
It is not.

Reply:
Just realised (d'oh) that the `parse_datetime` function is defined in this module! I think it'd be worth moving it to utils and using it every time we parse a datetime?

Reply:
(_Unless_ we want to explicitly error in the case where the precision isn't as expected for a given field?)

Reply:
Not all date time values are the same format. There would need to be a `key` passed along to some shared function (like done previously), but I'm not a fan of that—or there could be `n` different parse date time value functions, which I'm also not a fan of.

I'll keep it as it is to avoid too early DRY'ing it out.